### PR TITLE
Merge main into next branch

### PR DESCRIPTION
**Pull request by tylerbutler**
This PR merges main (commit 73929f265cea45f619e5dd1f7e9b0f16a973fa99) into the next branch. After the merge, there were several issues that needed to be fixed manually in order to get the build to succeed.

1. Updated versions of protocol-defs, container-defs, and driver-defs, as well as several server packages. Some bumps happened in both branches, so this PR takes the highest version of the dependencies from either branch.
2. Changes from the next-branch PR #10192 required changes from the PR #10447, which has not yet been merged. I squash-merged the changes from the open PR.

### CODE REVIEW

**Review by skylerjokiel — APPROVED** [review]
I took a high level cruse through and it looks good.

If there is something you would like feedback on specifically from Mark or Scott it might be best to @mention them in a comment since this PR is pretty huge.

File: api-report/container-loader.api.md [hunk 1]
```diff
@@ -43,9 +43,12 @@ import { TelemetryLogger } from '@fluidframework/telemetry-utils';
 
 // @public (undocumented)
 export enum ConnectionState {
+    CatchingUp = 1,
     Connected = 2,
+    // @deprecated (undocumented)
     Connecting = 1,
-    Disconnected = 0
+    Disconnected = 0,
+    EstablishingConnection = 3
```
[the remark this answers is not in the document]
Comment (scottn12) [reply api-report/container-loader.api.md:51]: @skylerjokiel, that's intentional. We are essentially renaming `Connecting` to `CatchingUp`, so they share the same value here. You can see https://github.com/microsoft/FluidFramework/pull/10416 for more context.

**Review by markfields — APPROVED** [review]
🎂 LGTM, I compared my PR #10447 to this and that part looks good.  I'll still follow up with a BREAKING.md change for those changes, btw.

File: packages/runtime/container-runtime-definitions/src/test/types/validateContainerRuntimeDefinitionsPrevious.ts [hunk 2]
```diff
@@ -6,7 +6,6 @@
  * THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
  * Generated by fluid-type-validator in @fluidframework/build-tools.
  */
-/* eslint-disable max-lines */
```
[the remark this answers is not in the document]
Comment (tylerbutler) [reply packages/runtime/container-runtime-definitions/src/test/types/validateContainerRuntimeDefinitionsPrevious.ts:9]: I fixed this in a main branch commit. The generated files no longer emit the disable comment, and there's a straightforward way to add type-compat-specific lint overrides in the config if needed.

**Comment by tylerbutler** [comment]
@skylerjokiel @markfields @scottn12 Thanks for the reviews. Merged!

**Pull request merged**